src/map/utils/charutils.cpp

```diff
@@ -2186,9 +2186,11 @@ namespace charutils
         }
         if (equipSlotID == SLOT_MAIN || equipSlotID == SLOT_RANGED || equipSlotID == SLOT_SUB)
         {
-            // If the weapon ISN'T an wind based instrument or an string based instrument
-            if (((CItemWeapon*)PItem)->getSkillType() != SKILL_STRING_INSTRUMENT && ((CItemWeapon*)PItem)->getSkillType() != SKILL_WIND_INSTRUMENT)
+            if (!PItem || !PItem->isType(ITEM_EQUIPMENT) ||
```

Comment:
```c++
auto PWeapon = dynamic_cast<CItemWeapon*>(PItem)
if (PWeapon && PWeapon->getSkillType() != SKILL_STRING_INSTRUMENT && PWeapon->getSkillType() != SKILL_WIND_INSTRUMENT)
```
That's what was meant on discord about casting. I haven't tested that and it was suggested to me by a 3rd party

Reply:
This would work too; minor change on such a small fix but a bit cleaner.

Reply:
+1 on this recommendation, a rare and beautiful possible use of `dynamic_cast`